cirq/study/sweep_trial_result.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Defines sweep trial results."""
+
+from typing import Dict, Any, TYPE_CHECKING, List, Set
+
+from cirq import value
+from cirq.study import resolver, trial_result
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import
+    import cirq
+
+
+@value.value_equality(unhashable=True)
+class SweepTrialResult:
+    """The combination of multiple TrialResults with potentially different
+       set of parameters.
+
+    Attributes:
+        trial_results: A list of TrialResults.
+    """
+
+    def __init__(self,
+                 trial_results: List[trial_result.TrialResult] = None) -> None:
+        self.trial_results = trial_results if trial_results is not None else []
+        self._index = {}  # type: Dict[str, Set[int]]
+        for i, result in enumerate(self.trial_results):
+            for key, value in result.params.param_dict.items():
+                self._index.setdefault(key + str(value), set()).add(i)
```

Comment:
Why not use the tuple `(key, value)` as the index key? The string concatenation is ambiguous.